apps/ledger-live-mobile/src/components/SelectDevice2/index.tsx

```diff
@@ -101,12 +101,18 @@ export default function SelectDevice({
       if (e.type === "remove") setDevice(undefined);
       if (e.type === "add") {
         const { name, deviceModel, id, wired } = e;
-        setDevice({
-          deviceName: name,
-          modelId: deviceModel?.id,
-          deviceId: id,
-          wired,
-        } as Device);
+
+        setDevice((maybeDevice: Device | undefined) => {
+          return (
+            maybeDevice ||
```

Comment:
Maybe just add a comment to explain why we're not replacing the (hid or httpdebug) device if one has already been set and it was not explictly removed (with a `remove` event)